bazel/EnvoyMobileTestSuite.kt

```diff
@@ -1,68 +1,31 @@
 package io.envoyproxy.envoymobile.bazel
 
-import java.io.File
-import java.net.URLClassLoader
-import java.util.zip.ZipFile
+import io.github.classgraph.ClassGraph
 import junit.framework.JUnit4TestAdapter
 import junit.framework.TestSuite
 import org.junit.runner.RunWith
 
-/**
- * This is class is taken from https://stackoverflow.com/questions/46365464/how-to-run-all-tests-in-bazel-from-a-single-java-test-rule
- *
- * Translated to Kotlin and slightly modified
- *
- * Requirements:
- * 1. This only allows tests to be run if the package is within `io.envoyproxy.envoymobile`
- * 2. This requires at least one test to be defined in the test target
- */
 @RunWith(org.junit.runners.AllTests::class)
 object EnvoyMobileTestSuite {
-  private const val TEST_SUFFIX = "Test"
-  private const val CLASS_SUFFIX = ".class"
+  private val junitTestAnnotation = org.junit.Test::class.java.name
 
   @JvmStatic
   fun suite(): TestSuite {
     val suite = TestSuite()
-    val classLoader = Thread.currentThread().contextClassLoader as URLClassLoader
-    val testAdapters = mutableListOf<JUnit4TestAdapter>()
-    // The first entry on the classpath contains the srcs from java_test
-    val classesInJar = findClassesInJar(File(classLoader.urLs[0].path))
-    for (clazz in classesInJar) {
-      val name = Class.forName(clazz)
-      val testAdapter = JUnit4TestAdapter(name)
-      testAdapters.add(testAdapter)
-    }
 
-    if (testAdapters.isEmpty()) {
-      throw NoTestFoundException("Unable to find any tests in test target")
-    }
+    val scan = ClassGraph()
+      .disableModuleScanning()
+      .enableAnnotationInfo()
+      .enableMethodInfo()
+      .ignoreClassVisibility()
+      .acceptPackages("io.envoyproxy")
```

Comment:
Double check that this catches all of the test classes that the original implementation used to find.

Reply:
It's not catching them all, thanks for pointing it out. Digging in.

Reply:
I'm removing the filtering, as it needs to pick up a wide variety of them.

Reply:
I'll add the other ones as well.

Reply:
Should be good now.